source/extensions/filters/http/grpc_json_transcoder/json_transcoder_filter.cc

```diff
@@ -537,6 +634,86 @@ bool JsonTranscoderFilter::readToBuffer(Protobuf::io::ZeroCopyInputStream& strea
   return false;
 }
 
+void JsonTranscoderFilter::createHttpBodyEnvelope(Buffer::Instance& data, uint64_t content_length) {
+  // Manually encode the grpc and protobuf envelope for the body.
+  // See https://developers.google.com/protocol-buffers/docs/encoding#embedded for wire format.
+
+  // Embedded messages are treated the same way as strings (wire type 2).
+  constexpr uint32_t ProtobufLengthDelimitedField = 2;
+
+  std::string request_prefix = request_prefix_.toString();
```

Comment:
Not quite understand what request_prefix is for?  could not add some examples in the comment